pkg/xds/generator/ingress_generator.go

```diff
@@ -112,6 +118,28 @@ func (i IngressGenerator) generateLDS(
 	return inboundListenerBuilder.Build()
 }
 
+func tagsFromTargetRef(targetRef common_api.TargetRef) (tags.Tags, bool) {
```

Comment:
Maybe I can refactor this in another PR? It might involve quite a bit of moving code around since I think the `tags.Tags` `struct` probably needs to be split up into non Envoy specific and Envoy specific functionality and moved to `pkg/core/xds` out of `pkg/envoy/xds`.